src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php

```
@@ -0,0 +1,297 @@
+
+    /**
+     * @var array Supported types
+     */
+    public static $TYPES = array('primary','secondary','tertiary','quaternary','quinary','link');
+
+    /**
+     * @var array Supported sizes
+     */
+    public static $SIZES = array('xl','lg','default','sm','xs');
+
+    /**
+     * @param boolean $linkNewWindow
+     *
+     * @return ButtonPagePart
+     */
+    public function setLinkNewWindow($linkNewWindow)
+    {
+        $this->linkNewWindow = $linkNewWindow;
+
+        return $this;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function isLinkNewWindow()
+    {
+        return $this->linkNewWindow;
+    }
+
+    /**
+     * @param string $linkText
+     *
+     * @return ButtonPagePart
+     */
+    public function setLinkText($linkText)
+    {
+        $this->linkText = $linkText;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getLinkText()
+    {
+        return $this->linkText;
+    }
+
+    /**
+     * @param string $linkUrl
+     *
+     * @return ButtonPagePart
+     */
+    public function setLinkUrl($linkUrl)
+    {
+        $this->linkUrl = $linkUrl;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getLinkUrl()
+    {
+        return $this->linkUrl;
+    }
+
+    /**
+     * Set type
+     *
+     * @param string $type
+     * @return ButtonPagePart
+     */
+    public function setType($type)
+    {
+        if (!in_array($type, self::$TYPES)) {
+            $type = 'primary';
+        }
+        $this->type = $type;
+
+        return $this;
+    }
+
+    /**
+     * Get type
+     *
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * Set size
+     *
+     * @param string $size
+     * @return ButtonPagePart
+     */
+    public function setSize($size)
+    {
+        if (!in_array($size, self::$SIZES)) {
+            $size = 'default';
```

imho this should throw an exception(fast) or trigger a validation error (clean)